### ISSUE TITLE
Failed to compile accessors for plugin org.hidetake.ssh

### ISSUE THREAD

**Issue by apomelov** [gradle/kotlin-dsl#612]
Build fails when I'm trying to use [hidetake SSH plugin](https://plugins.gradle.org/plugin/org.hidetake.ssh).

### Expected Behavior
I expect the SSH plugin to work (at least build script should compile).

### Current Behavior
As soon as I add it to my build script, build script is failing to compile.

### Context
I'm going to use SSH plugin to create a deploy task that will upload a new build from CI to a staging server.

### Steps to Reproduce
* In an empty directory:
```
gradle init
gradle wrapper
echo "" > settings.gradle
rm build.gradle
```
* Create a `build.gradle.kts` file:
```
repositories {
    jcenter()
}
plugins {
    kotlin("jvm").version("1.1.60")
    id("org.hidetake.ssh").version("2.9.0")
}
dependencies {
    compile(kotlin("stdlib-jre8", "1.1.60"))
}
```
* Run `./gradlew build` ->
```
$ ./gradlew build
e: Supertypes of the following classes cannot be resolved. Please make sure you have the required dependencies in the classpath:
    class org.hidetake.groovy.ssh.core.Remote, unresolved supertypes: FieldHelper, FieldHelper, FieldHelper, FieldHelper, FieldHelper, FieldHelper, FieldHelper, FieldHelper, FieldHelper


FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'reproduce'.
> Could not open cache directory ct3clo8evxdcnnwaf3l1bs7lw (C:\Users\apomelov\.gradle\caches\4.3\gradle-kotlin-dsl-accessors\ct3clo8evxdcnnwaf3l1bs7lw).
   > Failed to compile accessors.
```

I've posted this problem to the slack channel, Rodrigo B. de Olivera said, that it may be some issue in the Kotlin compiler. Here is the [thread](https://kotlinlang.slack.com/conversation/C19FD9681/p1511531184000374).

### My Environment
```
$ gradle -version

------------------------------------------------------------
Gradle 4.3
------------------------------------------------------------

Build time:   2017-10-30 15:43:29 UTC
Revision:     c684c202534c4138b51033b52d871939b8d38d72

Groovy:       2.4.12
Ant:          Apache Ant(TM) version 1.9.6 compiled on June 29 2015
JVM:          1.8.0_131 (Oracle Corporation 25.131-b11)
OS:           Windows 10 10.0 amd64
```

Cannot provide a build scan due to configuration error.

**Comment by bamboo** [comment]
Thanks for the report, @apomelov.

**Comment by bamboo** [comment]
Probably related to #611

**Comment by bamboo** [comment]
Blocked by #611.

**Comment by eskatos** [comment]
After re-assessment, this issue is not #611 but was fixed in Gradle Kotlin DSL 0.14 that shipped with Gradle 4.5.

Closing

**Comment by ingokegel** [comment]
No, the same problem occurs with Gradle 4.6.

We're using the SSH plugin with Gradle Kotlin DSL. It's not straight-forward, you have to insert a dynamic proxy to call methods in the RunHandler interface of the plugin. The plugin author doesn't use static typing everywhere and methodMissing is used to handle API calls.

Somebody will have to rewrite this plugin in Kotlin, I think.

**Comment by eskatos** [comment]
@ingokegel I can't reproduce with Gradle >= 4.5: https://asciinema.org/a/dnqcOgJlu2QLxZNsRczv0KvcD

**Comment by ingokegel** [comment]
The problem does not occur when you apply the plugin, only when you call methods on the ssh extension.

    ssh.remotes {
      myServer {
        host = '192.168.1.123'
      }
    }

**Comment by eskatos** [comment]
@ingokegel I see
This issue is about a script compilation issue, what you describe must be something else.

This works for me:
```kotlin
plugins {
    id("org.hidetake.ssh") version "2.9.0"
}
ssh.remotes {
    create("myServer") {
        host = "192.168.1.123"
    }
}
```

Could you please open another issue for your other problem if need be?

**Comment by ingokegel** [comment]
Sorry, I pasted Groovy code, I just updated it with Kotlin code above. I'm not sure it makes any sense to try to get this plugin to work with Kotlin because there are so many problems with it. But this error may be interesting for Groovy interoperability in general.

**Comment by eskatos** [comment]
If this plugin's API needs some refinements for better usability from Kotlin, opening issues on the [plugin repository](https://github.com/int128/gradle-ssh-plugin) would make sense.

**Comment by ingokegel** [comment]
OK, I've prepared a same that produces the "Supertypes of the following classes cannot be resolved" error on compilation:

	import org.hidetake.groovy.ssh.core.Remote
	import org.hidetake.groovy.ssh.core.RunHandler
	import org.hidetake.groovy.ssh.session.SessionHandler

	plugins {
		id("org.hidetake.ssh") version "2.9.0"
	}

	ssh.run(delegateClosureOf<RunHandler> {
		session((ssh.remotes as Map<*, *>)["server"]!! as Remote, delegateClosureOf<SessionHandler> {
			get(hashMapOf("from" to "remoteFile.txt", "into" to "localFile.txt"))
		})
	})

**Comment by eskatos** [comment]
@ingokegel nice!
It looks related to Groovy traits.
Could you please open another issue, this is a different problem than described here.

**Comment by ingokegel** [comment]
Done #763

**Comment by eskatos** [comment]
Thank you!
